### ISSUE TITLE
Use HTTP/1.1 by default when contacting github.com

### ISSUE THREAD

**Issue by charles-dyfis-net** [NixOS/nix#6763]
**Is your feature request related to a problem? Please describe.**

When contacting GitHub support for a HTTP 503 error repeatedly encountered when Nix is downloading a large tarball from github.com in a resource-constrained container, it came up that HTTP/2 is not an officially supported interface; only HTTP/1.1 is guaranteed to work at this time.

**Describe the solution you'd like**

fetchFromGitHub and similar should force HTTP/1.1 until GitHub officially supports HTTP/2.

**Describe alternatives you've considered**

Our immediate workaround was adding `--option http2 false` on the specific `nix-build` operation hitting this error. However, this is a much larger hammer than necessary; and insofar as `cache.nixos.org` is optimized for HTTP/2, an unfortunate one.

**Additional context**

In contacting GitHub support about this, you may wish to reference support ticket #1674959. The statement therein that HTTP/2 is not yet supported was made on June 30th, 2022.

**Comment by edolstra** [comment]
If GitHub doesn't "support" HTTP/2, then their server shouldn't respond to HTTP/2 requests. Surely they don't expect every client to have a blacklist/whitelist of servers that appear to handle HTTP/2 but are actually broken.

**Comment by charles-dyfis-net** [comment]
I agree, and encouraged them to stop providing an unsupported interface in
the aforementioned support ticket. However, they haven't yet -- and GitHub
is already large and influential enough to justify having a special-case
`fetchFromGitHub`; perhaps similar reasoning applies here.

**Comment by tomberek** [comment]
Closing as stale. Re-open if needed.
